modules/admin_manual/pages/configuration/server/security/hsmdaemon.adoc

```diff
@@ -0,0 +1,514 @@
+mTM2gZlUnGKXN7c4kaBPFt1IfjjVYu7pvj3B2uxUf4GywuSuWGWnAy89FqeXteRV
+jwIDAQAB
+-----END PUBLIC KEY-----
+----
+
+==== Testing Data Encryption
+
+For testing data encryption, run the `hsmdaemon encrypt` command, as in the following example.
+
+[source,console]
+----
+The first argument is the "Id:" value from running the genkey command above.
+$ sudo hsmdaemon encrypt 9bac3719-2b8d-11e9-aeab-0242b5ece4c3 Zm9vYmFy
+----
+
+If successful, you should see output similar to the below example.
+
+[source,console]
+----
+{"level":"debug","ts":"2019-03-20T12:43:40.540+0100","caller":"hsmdaemon/keymanager.go:27","msg":"initialize pkcs11 module","module":"/usr/lib/softhsm/libsofthsm2.so"}
+{"level":"debug","ts":"2019-03-20T12:43:40.545+0100","caller":"hsmdaemon/keymanager.go:205","msg":"openHSMSession","slotID":858597139}
+{"level":"info","ts":"2019-03-20T12:43:40.549+0100","caller":"hsmdaemon/keymanager.go:621","msg":"Fetching private key","keyID":"9bac3719-2b8d-11e9-aeab-0242b5ece4c3"}
+{"level":"debug","ts":"2019-03-20T12:43:40.549+0100","caller":"hsmdaemon/keymanager.go:641","msg":"Got uuid","string":"13d34146-4b02-11e9-adbd-0023ae27c404"}
+WcezVb2N6bF8wlDooKZcmFn3tZgoIpoFGx6wQetx9sp1nK7JW2Y4OKt7P+0VKKlFO7yXaffVDD2Q6jZZCQukQVRV1zJrwbI9xU3YlOAwJFPP+WM/dZ1vdUwi7L05wq8UpL13LJWlMkvd1eIqKJS7apMnFk2hbnxXP6UKZmI++1tXvqbAc6fwhcB5J+JG6lmS4RwnD+eJC3dq5t00zzdI6vuIM/y3UT7ESklmHl5bKl+N+d6yk6qLxnFnIJweL+M3Tf13+XPNAh5JxZpheJPvN3oL28uX76aizy4BCLnRgQ/ryUQeDF+a4zNF22sMwBh4Pt46KrYGNDZAnQpVzmkrZQ==
+----
+
+==== Testing Showing Keys
+
+To show an existing key, use the `showkey` command with the key’s id, as in the following example.
+
+[source,console]
+----
+sudo hsmdaemon showkey 9bac3719-2b8d-11e9-aeab-0242b5ece4c3
+----
+
+////
+==== Testing Data Decryption
+
+TODO.
+
+==== Testing Key Deletion
+
+TODO.
+////
+
+==== [optional] Configure Other Options
+
+For more options see the self-documented default config file `hsmdaemon.toml`.
+
+==== Run in Foreground During Setup
+
+During ownCloud config you might want to run the service in the foreground to see what is going on:
+
+[source,console]
+----
+$ ./hsmdaemon
+{
+    "level": "info",
+    "ts": "2019-02-14T09:32:59.081+0100",
+    "caller": "hsmdaemon/hsmdaemon.go:146",
+    "msg": "Server listening",
+    "host": "localhost",
+    "port": 8513,
+    "version": "0.0.7",
+    "build": "2019-02-08T10:47:55+00:00"
+}
+----
+
+NOTE: The above console output is formatted for readability.
+
+=== ownCloud Setup
+
+==== Shut Down Apache
+
+If anyone accesses ownCloud while encryption is enabled, it will automatically generate the keys. 
+To prevent this, shut down Apache until encryption is appropriately configured.
+
+==== Patch ownCloud
+
+NOTE: The PR for HSM support https://github.com/owncloud/core/pull/34205[is in review].
+
+For now, we assume a fresh ownCloud installation. 
+Patch the encryption app, by following the example below.
+
+[source,console]
+----
+$ cd /path/to/owncloud/installation
```

Comment:
Proposal:
`$ cd /path/to/owncloud/installation` —> `$ cd /path/to/owncloud_root/`
Else it could be understood to change to directory installation